src/main/java/games/strategy/util/memento/MementoImportException.java

```diff
@@ -0,0 +1,33 @@
+package games.strategy.util.memento;
+
+/**
+ * A checked exception that indicates an error occurred while importing a memento to its originator.
+ *
+ * <p>
+ * Instances of this class are thread safe.
+ * </p>
+ */
+public final class MementoImportException extends Exception {
+  private static final long serialVersionUID = -3885032648502798651L;
+
+  /**
+   * Initializes a new instance of the {@code MementoImportException} class with the specified detail message and no
+   * cause.
+   *
+   * @param message The detail message; may be {@code null}.
```

Comment:
I agree that non-`null` should be assumed to be the default, and as I mentioned someplace else, I normally don't document nullity at all because I tend to prefer using one of the many nullity annotation libraries available.  I will go ahead and remove all documentation for non-`null` parameters.

Actually, I'd prefer to not even have to add Javadocs for `null`-able parameters/return values.  Do you think introducing the `javax.annotation.Nullable` annotation in our codebase would be worthwhile?

> Considering this further, I would personally not allow for the message string to either be null or empty.

In general, I agree with you.  However, the API for `Exception` constructors is pretty old and the use of `null` arguments for these parameters seems well established.  I don't think the case we have to worry about is a dev explicitly passing `null` for one of these parameters because they didn't realize an alternate constructor was available.  Rather, I think the danger is from passing through values obtained from other sources.  For example, if someone were to do this (totally contrived, but I hope it makes my point):

```java
try {
  // . . .
} catch (final IOException e) {
  throw new MementoImportException(e.getMessage());
}
```

the dev would have to check the argument before calling an appropriate constructor because `Exception#getMessage()` can return `null`.  That is, the dev would be forced to jump through hoops like this:

```java
try {
  // . . .
} catch (final IOException e) {
  final String message = e.getMessage();
  if (message != null) {
    throw new MementoImportException(message);
  } else {
    throw new MementoImportException();
  }
}
```

Again, I agree with your philosophy about generally not accepting `null` when designing new APIs.  However, as I stated above, `Exception` is not new, and we're effectively extending its API rather than defining a new one.